app/controllers/users/registrations_controller.rb

```diff
@@ -36,17 +36,14 @@ def index
   def show
     @user = User.find(params[:id])
 
-    if @user.blank?
-      flash[:error] = 'Sorry, no such user.'
-      redirect_to projects_path
-      return
-    end
-
     if !@user.is_visible_to_user?(current_user)
       flash[:error] = 'Sorry, no such user.'
       redirect_to projects_path
-      nil
     end
+
+  rescue ActiveRecord::RecordNotFound
+    flash[:error] = 'Sorry, no such user.'
+    redirect_to projects_path
```

Comment:
I've changed the code to fail gracefully if `User.find(params[:id])` does not find a user as raises an error.